libraries/lib-dynamic-range-processor/DynamicRangeProcessorUtils.cpp

```diff
@@ -0,0 +1,125 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  DynamicRangeProcessorUtils.cpp
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#include "DynamicRangeProcessorUtils.h"
+#include <optional>
+#include <regex>
+#include <stdexcept>
+#include <unordered_map>
+
+namespace DynamicRangeProcessorUtils
+{
+namespace
+{
+
+template <typename Struct> struct SettingDescription
+{
+   double Struct::*const mem {};
+   const char* const key {};
+};
+
+static const std::vector<SettingDescription<CompressorSettings>>
```

Comment:
nitpick: could be a std::array if the size of container is known in advance

Reply:
Yeah, but then you have to count your number of presets and update that size - more human work `:P`. (What you suggest would have been fine too, of course.)